src/proto/streams/send.rs

```diff
@@ -155,14 +155,15 @@ impl Send {
     }
 
     pub fn schedule_cancel(&mut self, stream: &mut store::Ptr, task: &mut Option<Task>) {
+        trace!("schedule_cancel; {:?}", stream.id);
         if stream.state.is_closed() {
             // Stream is already closed, nothing more to do
             return;
         }
 
         stream.state.set_canceled();
 
-        self.reclaim_capacity(stream);
+        self.prioritize.reclaim_reserved_capacity(stream);
```

Comment:
Without this change, the implicit cancel was clearing the capacity of the buffered data frame, and it was hanging again.